Pack-Smart/pack-smart-fe
# Feature/patch me if you can

## What's this PR do?
- Create 'UpdateDetailsModal`
- Make action and reducer for `EDIT_TRIP_DETAILS`
- PATCH request TripDetails changes
- PATCH request item quantity and check mark changes

## Where should the reviewer start?
- `UpdateDetailsModals.js`, `actions.js`, `packingList.js` (in reducers), `apiCalls.js` and `StaticItem.js`

## How should this be manually tested?
- Change the trip details to see them reflected on the packing list view and on refresh
- Update the quantity or toggle the check mark and it will be preserved on refresh

## What are the relevant tickets?
- Closes #86 
- Closes #88 
- Closes #91
- Closes #93 

## Did you update the README?
- I did not!
